### ISSUE TITLE
Undocumented theme option `defaultMode` missing from TypeScript definitions

### ISSUE THREAD

**Issue by oskari** [grommet/grommet#4607]
### Expected Behavior

`defaultMode` is used in [Grommet component](https://github.com/grommet/grommet/blob/3b912b5023cb7ec381cdecb2f0391a203de498d3/src/js/components/Grommet/Grommet.js#L43) along with `themeMode` prop to determine initial theme mode between dark and light themes. As this property is used by Grommet Theme Designer, Grommet HPE Theme and in HPE Design System I would expect it to be documented and included in the TypeScript definitions.

### Actual Behavior

`defaultMode` is not documented and not included in TS definisions, resulting in an error when used

- Grommet version: 2.15.2

**Comment by ShimiSun** [comment]
In the spirit of [Hacktoberfest](https://hacktoberfest.digitalocean.com/), would you like to make a contribution?

**Comment by oskari** [comment]
Yeah I took a look at this and I guess I poorly worded this issue at first. It's not actually TypeScript issue per se,  but more like missing documentation all together. 

Is the `theme` prop that is passed to `Grommet` -component documented anywhere else other than [wiki?](https://github.com/grommet/grommet/wiki/Grommet-v2-theming-documentation). There's [few other fields](https://github.com/grommet/grommet-theme-designer/blob/3eb434c1713b9598afb00ec62088d18802fe50aa/src/theme.js#L5-L8) in the themes generated by [Grommet Theme Designer](https://github.com/grommet/grommet-theme-designer) that are not used by the looks of it, I guess they're more of a metadata? 

Should `defaultMode` be Grommet component property instead of being a property in the `theme` object? Wouldn't that be more in line of the current component design?

**Comment by ericsoderberghp** [comment]
You are correct about some of the properties included by the theme designer are metadata used by the theme designer.

`defaultMode` allows the theme to specify a default mode that is used when it isn't specified via `<Grommet themeMode={} />`.  This allows the theme to define a preferred themeMode.

**Comment by jcfilben** [comment]
Sounds like there is two things we should do here:
- [ ] Add `defaultMode` to the documentation on the [grommet-site](https://github.com/grommet/grommet-site) repo
- [ ] Add `defaultMode` to the typescript definitions in `themes/base.d.ts`
